### PR TITLE
Fix issue with transitive dependencies failure.

### PR DESCRIPTION
**Pull request by matejonnet**

### CODE REVIEW

**Comment by matejonnet** [comment]
@michalszynkiewicz Please take a look if it is ok to add this extra condition [1]. It is required to prevent infinite loop which occurred in case of failed transitive dependencies.

1: https://github.com/project-ncl/pnc/pull/1423/commits/f00aabe17eeb52d355735069f232d2426c54447c#diff-b094550a9072e565e980eab02bab41cbR453

File: build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/DefaultBuildCoordinator.java [hunk 1]
```diff
@@ -447,8 +447,10 @@ private boolean isFinished(BuildSetTask buildSetTask) {
     private void handleErroneousFinish(BuildTask failedTask) {
         BuildSetTask taskSet = failedTask.getBuildSetTask();
         if (taskSet != null) {
+            log.debug("Finishing tasks in set {}, after failedTask {}.", taskSet, failedTask);
             taskSet.getBuildTasks().stream()
                     .filter(t -> isDependentOn(failedTask, t))
+                    .filter(t -> !t.getStatus().isCompleted())
```
Comment (michalszynkiewicz) [inline build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/DefaultBuildCoordinator.java]: is this line the real change?

Do you know how it happen that a build task was completed and had a failed dependency?

Comment (matejonnet) [inline build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/DefaultBuildCoordinator.java]: Test scenario I added has the dependency chain.
A->B->C, where C is failing and B fall into infinite loop.

Yes that extra filter is a real change.

Comment (michalszynkiewicz) [inline build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/DefaultBuildCoordinator.java]: I get it now, thanks

**Comment by michalszynkiewicz** [comment]
:+1:

**Comment by matejonnet** [comment]
Retest this please.

**Pull request merged**